lib/fca_api/connection.rb

```diff
@@ -0,0 +1,55 @@
+module FcaApi
+  class Connection
+    attr_reader :domain, :raw_connection, :timeout
+    attr_reader :retries, :api_email, :api_key
+    delegate :headers, :post, to: :raw_connection
+
+    def initialize(domain)
+      @domain = domain
+      @api_email = ENV.fetch('FCA_API_EMAIL')
+      @api_key = ENV.fetch('FCA_API_KEY')
+      @retries = ENV.fetch('FCA_API_MAX_RETRIES').to_i
+      @timeout = ENV.fetch('FCA_API_TIMEOUT').to_i
+      configure_raw_connection
+    end
+
+    def configure_raw_connection
```

Comment:
do we need to expose this method in the Connection public interface?

Reply:
good point, have privatised it.